config/passport.js

```diff
@@ -0,0 +1,79 @@
+const GoogleStrategy = require('passport-google-oauth').OAuth2Strategy;
+const LocalStrategy = require('passport-local').Strategy;
+const BearerStrategy = require('passport-http-bearer').Strategy;
+const configAuth = require('./settings');
+
+module.exports = function(passport, models) {
+
+	const localUser = models.user_account;
+
+	passport.serializeUser(function(user, done){
+		done(null, user);
+	});
+
+	passport.deserializeUser(function(user, done){
+		done(null, user);
+	});
+
+	passport.use(new GoogleStrategy({
+			clientID: configAuth.googleAuth.clientID,
+			clientSecret: configAuth.googleAuth.clientSecret,
+			callbackURL: configAuth.googleAuth.callbackURL,
+			access_type: configAuth.googleAuth.access_type
+		},
+	  	function(req, accessToken, refreshToken, profile, done) {
+	  		process.nextTick(function(){
+	  			localUser.findOrCreate({where: {
+	  				email: profile.emails[0].value,
+	  				provider: 'google'
+	  			}, defaults:{
+	  				email: profile.emails[0].value,
+	  				provider: 'google',
+	  				google_token: accessToken,
+	  				google_id: profile.id,
+	  				name: profile.displayName
+	  			}})
+	  			.spread((user, created)=> {
+	  				if(!created && !user){
+	  					return done(null, false, {'error': 'User with that email already exists'});
```

Comment:
what's the purpose of the first argument then?
typically in js callback functions, the first argument indicates `error`, which is why it's `null` in most cases (error-free).
since you are passing in the error object in the 3rd argument, then what's the purpose of the first?